qiskit_experiments/framework/base_experiment.py

```diff
@@ -194,11 +220,16 @@ def run(
             QiskitError: if experiment is run with an incompatible existing
                          ExperimentData container.
         """
-        if backend is None:
-            experiment = self
-        else:
+        if backend is not None or isinstance(analysis, BaseAnalysis):
+            # Make a copy to update analysis or backend if one is provided at runtime
             experiment = self.copy()
-            experiment._set_backend(backend)
+            if backend:
+                experiment._set_backend(backend)
+            if isinstance(analysis, BaseAnalysis):
```

Comment:
```suggestion
            if not isinstance(analysis, bool) and analysis != "default":
```
this is more precise since we have validation in setter? i.e. with current implementation we can set whatever class instance and it passes without error. just nothing happens.

Reply:
I will add a separate deprecation warning for boolean values, then we only need to check for `"default"` and everything else should be valid for `exp.analysis` setter.